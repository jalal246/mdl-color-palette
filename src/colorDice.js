import * as COLORS from "./colors";

const collections = [
  "RED",
  "PINK",
  "PURPLE",
  "DEPP_PURPLE",
  "INDIGO",
  "BLUE",
  "LIGHT_BLUE",
  "CYAN",
  "TEAL",
  "GREEN",
  "LIGHT_GREEN",
  "LIME",
  "YELLOW",
  "AMBER",
  "ORANGE",
  "DEEP_ORANGE",
  "BROWN",
  "GREY",
  "BLUE_GREY",
];

function dice(min, max) {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

function colorDice() {
  return collections[dice(0, collections.length - 1)];
}

export default function colorDiceGen({ isLight = false, isDark = false } = {}) {
  const mainSelectedColor = COLORS[colorDice()];

  let from = 0;
  let to = mainSelectedColor.length - 1;

  if (isLight) {
    to = 4;
  } else if (isDark) {
    from = 4;
  }

  return mainSelectedColor[dice(from, to)];
}
