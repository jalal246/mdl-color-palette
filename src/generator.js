// Used once to generate json file

import fs from "fs";
import * as colors from "./colors";

const colorKeys = Object.keys(colors);

const numbers = [
  "50",
  "100",
  "200",
  "300",
  "400",
  "500",
  "600",
  "700",
  "800",
  "900",
];

const allPallet = {};

(function gen() {
  colorKeys.forEach((colorKey) => {
    const name = colorKey.toLowerCase().replace("_", "-");

    colors[colorKey].forEach((colorHex, i) => {
      allPallet[`${name}-${numbers[i]}`] = colorHex;
    });
  });

  fs.writeFileSync("./src/colors.json", JSON.stringify(allPallet));
})();
