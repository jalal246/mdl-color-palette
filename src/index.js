/* eslint-disable import/prefer-default-export */
import colorDice from "./colorDice";

export { colorDice };
