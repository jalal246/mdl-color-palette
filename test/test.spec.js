import * as colorsJson from "../src/colors.json";
import * as colorsArray from "../src/colors";

import colorDice from "../src/colorDice";

describe("Test all palette", () => {
  it("Tests palette generated in JSON", () => {
    expect(colorsJson).toMatchSnapshot();
  });
  it("Tests palette exists in row data", () => {
    expect(colorsArray).toMatchSnapshot();
  });
});

describe("Test randomness in color dice", () => {
  it("Gets random color", () => {
    expect(colorDice()).toBeTruthy();
  });

  it("Gets random light color", () => {
    expect(colorDice({ isLight: true })).toBeTruthy();
  });

  it("Gets random dark color", () => {
    expect(colorDice({ isDark: true })).toBeTruthy();
  });
});
