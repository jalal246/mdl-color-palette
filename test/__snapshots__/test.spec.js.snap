// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test all palette Tests palette exists in row data 1`] = `
Object {
  "AMBER": Array [
    "#fff8e1",
    "#ffecb3",
    "#ffe082",
    "#ffd54f",
    "#ffca28",
    "#ffc107",
    "#ffb300",
    "#ffa000",
    "#ff8f00",
    "#ff6f00",
  ],
  "BLUE": Array [
    "#e3f2fd",
    "#bbdefb",
    "#90caf9",
    "#64b5f6",
    "#42a5f5",
    "#2196f3",
    "#1e88e5",
    "#1976d2",
    "#1565c0",
    "#0d47a1",
  ],
  "BLUE_GREY": Array [
    "#eceff1",
    "#cfd8dc",
    "#b0bec5",
    "#90a4ae",
    "#78909c",
    "#607d8b",
    "#546e7a",
    "#455a64",
    "#37474f",
    "#263238",
  ],
  "BROWN": Array [
    "#efebe9",
    "#d7ccc8",
    "#bcaaa4",
    "#a1887f",
    "#8d6e63",
    "#795548",
    "#6d4c41",
    "#5d4037",
    "#4e342e",
    "#3e2723",
  ],
  "CYAN": Array [
    "#e0f7fa",
    "#b2ebf2",
    "#80deea",
    "#4dd0e1",
    "#26c6da",
    "#00bcd4",
    "#00acc1",
    "#0097a7",
    "#00838f",
  ],
  "DEEP_ORANGE": Array [
    "#fbe9e7",
    "#ffccbc",
    "#ffab91",
    "#ff8a65",
    "#ff7043",
    "#ff5722",
    "#f4511e",
    "#e64a19",
    "#d84315",
    "#bf360c",
  ],
  "DEPP_PURPLE": Array [
    "#ede7f6",
    "#d1c4e9",
    "#b39ddb",
    "#9575cd",
    "#7e57c2",
    "#673ab7",
    "#5e35b1",
    "#512da8",
    "#4527a0",
    "#311b92",
  ],
  "GREEN": Array [
    "#e8f5e9",
    "#c8e6c9",
    "#a5d6a7",
    "#81c784",
    "#66bb6a",
    "#4caf50",
    "#43a047",
    "#388e3c",
    "#2e7d32",
    "#1b5e20",
  ],
  "GREY": Array [
    "#fafafa",
    "#f5f5f5",
    "#eeeeee",
    "#e0e0e0",
    "#bdbdbd",
    "#9e9e9e",
    "#757575",
    "#616161",
    "#424242",
    "#212121",
  ],
  "INDIGO": Array [
    "#e8eaf6",
    "#c5cae9",
    "#9fa8da",
    "#7986cb",
    "#5c6bc0",
    "#3f51b5",
    "#3949ab",
    "#303f9f",
    "#283593",
    "#1a237e",
  ],
  "LIGHT_BLUE": Array [
    "#e1f5fe",
    "#b3e5fc",
    "#81d4fa",
    "#4fc3f7",
    "#29b6f6",
    "#03a9f4",
    "#039be5",
    "#0288d1",
    "#0277bd",
    "#01579b",
  ],
  "LIGHT_GREEN": Array [
    "#f1f8e9",
    "#dcedc8",
    "#c5e1a5",
    "#aed581",
    "#9ccc65",
    "#8bc34a",
    "#7cb342",
    "#689f38",
    "#558b2f",
    "#33691e",
  ],
  "LIME": Array [
    "#f9fbe7",
    "#f0f4c3",
    "#e6ee9c",
    "#dce775",
    "#d4e157",
    "#cddc39",
    "#c0ca33",
    "#afb42b",
    "#9e9d24",
    "#827717",
  ],
  "ORANGE": Array [
    "#fff3e0",
    "#ffe0b2",
    "#ffcc80",
    "#ffb74d",
    "#ffa726",
    "#ff9800",
    "#fb8c00",
    "#f57c00",
    "#ef6c00",
    "#e65100",
  ],
  "PINK": Array [
    "#fce4ec",
    "#f8bbd0",
    "#f48fb1",
    "#f06292",
    "#ec407a",
    "#e91e63",
    "#d81b60",
    "#c2185b",
    "#ad1457",
    "#880e4f",
  ],
  "PURPLE": Array [
    "#f3e5f5",
    "#e1bee7",
    "#ce93d8",
    "#ba68c8",
    "#ab47bc",
    "#9c27b0",
    "#8e24aa",
    "#7b1fa2",
    "#6a1b9a",
    "#4a148c",
  ],
  "RED": Array [
    "#ffebee",
    "#ffcdd2",
    "#ef9a9a",
    "#e57373",
    "#ef5350",
    "#f44336",
    "#e53935",
    "#d32f2f",
    "#c62828",
    "#b71c1c",
  ],
  "TEAL": Array [
    "#e0f2f1",
    "#b2dfdb",
    "#80cbc4",
    "#4db6ac",
    "#26a69a",
    "#009688",
    "#00897b",
    "#00796b",
    "#00695c",
    "#004d40",
  ],
  "YELLOW": Array [
    "#FFFDE7",
    "#FFF9C4",
    "#FFF59D",
    "#FFF176",
    "#FFEE58",
    "#FFEB3B",
    "#FDD835",
    "#FBC02D",
    "#F9A825",
    "#F57F17",
  ],
}
`;

exports[`Test all palette Tests palette generated in JSON 1`] = `
Object {
  "amber-100": "#ffecb3",
  "amber-200": "#ffe082",
  "amber-300": "#ffd54f",
  "amber-400": "#ffca28",
  "amber-50": "#fff8e1",
  "amber-500": "#ffc107",
  "amber-600": "#ffb300",
  "amber-700": "#ffa000",
  "amber-800": "#ff8f00",
  "amber-900": "#ff6f00",
  "blue-100": "#bbdefb",
  "blue-200": "#90caf9",
  "blue-300": "#64b5f6",
  "blue-400": "#42a5f5",
  "blue-50": "#e3f2fd",
  "blue-500": "#2196f3",
  "blue-600": "#1e88e5",
  "blue-700": "#1976d2",
  "blue-800": "#1565c0",
  "blue-900": "#0d47a1",
  "blue-grey-100": "#cfd8dc",
  "blue-grey-200": "#b0bec5",
  "blue-grey-300": "#90a4ae",
  "blue-grey-400": "#78909c",
  "blue-grey-50": "#eceff1",
  "blue-grey-500": "#607d8b",
  "blue-grey-600": "#546e7a",
  "blue-grey-700": "#455a64",
  "blue-grey-800": "#37474f",
  "blue-grey-900": "#263238",
  "brown-100": "#d7ccc8",
  "brown-200": "#bcaaa4",
  "brown-300": "#a1887f",
  "brown-400": "#8d6e63",
  "brown-50": "#efebe9",
  "brown-500": "#795548",
  "brown-600": "#6d4c41",
  "brown-700": "#5d4037",
  "brown-800": "#4e342e",
  "brown-900": "#3e2723",
  "cyan-100": "#b2ebf2",
  "cyan-200": "#80deea",
  "cyan-300": "#4dd0e1",
  "cyan-400": "#26c6da",
  "cyan-50": "#e0f7fa",
  "cyan-500": "#00bcd4",
  "cyan-600": "#00acc1",
  "cyan-700": "#0097a7",
  "cyan-800": "#00838f",
  "deep-orange-100": "#ffccbc",
  "deep-orange-200": "#ffab91",
  "deep-orange-300": "#ff8a65",
  "deep-orange-400": "#ff7043",
  "deep-orange-50": "#fbe9e7",
  "deep-orange-500": "#ff5722",
  "deep-orange-600": "#f4511e",
  "deep-orange-700": "#e64a19",
  "deep-orange-800": "#d84315",
  "deep-orange-900": "#bf360c",
  "default": Object {
    "amber-100": "#ffecb3",
    "amber-200": "#ffe082",
    "amber-300": "#ffd54f",
    "amber-400": "#ffca28",
    "amber-50": "#fff8e1",
    "amber-500": "#ffc107",
    "amber-600": "#ffb300",
    "amber-700": "#ffa000",
    "amber-800": "#ff8f00",
    "amber-900": "#ff6f00",
    "blue-100": "#bbdefb",
    "blue-200": "#90caf9",
    "blue-300": "#64b5f6",
    "blue-400": "#42a5f5",
    "blue-50": "#e3f2fd",
    "blue-500": "#2196f3",
    "blue-600": "#1e88e5",
    "blue-700": "#1976d2",
    "blue-800": "#1565c0",
    "blue-900": "#0d47a1",
    "blue-grey-100": "#cfd8dc",
    "blue-grey-200": "#b0bec5",
    "blue-grey-300": "#90a4ae",
    "blue-grey-400": "#78909c",
    "blue-grey-50": "#eceff1",
    "blue-grey-500": "#607d8b",
    "blue-grey-600": "#546e7a",
    "blue-grey-700": "#455a64",
    "blue-grey-800": "#37474f",
    "blue-grey-900": "#263238",
    "brown-100": "#d7ccc8",
    "brown-200": "#bcaaa4",
    "brown-300": "#a1887f",
    "brown-400": "#8d6e63",
    "brown-50": "#efebe9",
    "brown-500": "#795548",
    "brown-600": "#6d4c41",
    "brown-700": "#5d4037",
    "brown-800": "#4e342e",
    "brown-900": "#3e2723",
    "cyan-100": "#b2ebf2",
    "cyan-200": "#80deea",
    "cyan-300": "#4dd0e1",
    "cyan-400": "#26c6da",
    "cyan-50": "#e0f7fa",
    "cyan-500": "#00bcd4",
    "cyan-600": "#00acc1",
    "cyan-700": "#0097a7",
    "cyan-800": "#00838f",
    "deep-orange-100": "#ffccbc",
    "deep-orange-200": "#ffab91",
    "deep-orange-300": "#ff8a65",
    "deep-orange-400": "#ff7043",
    "deep-orange-50": "#fbe9e7",
    "deep-orange-500": "#ff5722",
    "deep-orange-600": "#f4511e",
    "deep-orange-700": "#e64a19",
    "deep-orange-800": "#d84315",
    "deep-orange-900": "#bf360c",
    "depp-purple-100": "#d1c4e9",
    "depp-purple-200": "#b39ddb",
    "depp-purple-300": "#9575cd",
    "depp-purple-400": "#7e57c2",
    "depp-purple-50": "#ede7f6",
    "depp-purple-500": "#673ab7",
    "depp-purple-600": "#5e35b1",
    "depp-purple-700": "#512da8",
    "depp-purple-800": "#4527a0",
    "depp-purple-900": "#311b92",
    "green-100": "#c8e6c9",
    "green-200": "#a5d6a7",
    "green-300": "#81c784",
    "green-400": "#66bb6a",
    "green-50": "#e8f5e9",
    "green-500": "#4caf50",
    "green-600": "#43a047",
    "green-700": "#388e3c",
    "green-800": "#2e7d32",
    "green-900": "#1b5e20",
    "grey-100": "#f5f5f5",
    "grey-200": "#eeeeee",
    "grey-300": "#e0e0e0",
    "grey-400": "#bdbdbd",
    "grey-50": "#fafafa",
    "grey-500": "#9e9e9e",
    "grey-600": "#757575",
    "grey-700": "#616161",
    "grey-800": "#424242",
    "grey-900": "#212121",
    "indigo-100": "#c5cae9",
    "indigo-200": "#9fa8da",
    "indigo-300": "#7986cb",
    "indigo-400": "#5c6bc0",
    "indigo-50": "#e8eaf6",
    "indigo-500": "#3f51b5",
    "indigo-600": "#3949ab",
    "indigo-700": "#303f9f",
    "indigo-800": "#283593",
    "indigo-900": "#1a237e",
    "light-blue-100": "#b3e5fc",
    "light-blue-200": "#81d4fa",
    "light-blue-300": "#4fc3f7",
    "light-blue-400": "#29b6f6",
    "light-blue-50": "#e1f5fe",
    "light-blue-500": "#03a9f4",
    "light-blue-600": "#039be5",
    "light-blue-700": "#0288d1",
    "light-blue-800": "#0277bd",
    "light-blue-900": "#01579b",
    "light-green-100": "#dcedc8",
    "light-green-200": "#c5e1a5",
    "light-green-300": "#aed581",
    "light-green-400": "#9ccc65",
    "light-green-50": "#f1f8e9",
    "light-green-500": "#8bc34a",
    "light-green-600": "#7cb342",
    "light-green-700": "#689f38",
    "light-green-800": "#558b2f",
    "light-green-900": "#33691e",
    "lime-100": "#f0f4c3",
    "lime-200": "#e6ee9c",
    "lime-300": "#dce775",
    "lime-400": "#d4e157",
    "lime-50": "#f9fbe7",
    "lime-500": "#cddc39",
    "lime-600": "#c0ca33",
    "lime-700": "#afb42b",
    "lime-800": "#9e9d24",
    "lime-900": "#827717",
    "orange-100": "#ffe0b2",
    "orange-200": "#ffcc80",
    "orange-300": "#ffb74d",
    "orange-400": "#ffa726",
    "orange-50": "#fff3e0",
    "orange-500": "#ff9800",
    "orange-600": "#fb8c00",
    "orange-700": "#f57c00",
    "orange-800": "#ef6c00",
    "orange-900": "#e65100",
    "pink-100": "#f8bbd0",
    "pink-200": "#f48fb1",
    "pink-300": "#f06292",
    "pink-400": "#ec407a",
    "pink-50": "#fce4ec",
    "pink-500": "#e91e63",
    "pink-600": "#d81b60",
    "pink-700": "#c2185b",
    "pink-800": "#ad1457",
    "pink-900": "#880e4f",
    "purple-100": "#e1bee7",
    "purple-200": "#ce93d8",
    "purple-300": "#ba68c8",
    "purple-400": "#ab47bc",
    "purple-50": "#f3e5f5",
    "purple-500": "#9c27b0",
    "purple-600": "#8e24aa",
    "purple-700": "#7b1fa2",
    "purple-800": "#6a1b9a",
    "purple-900": "#4a148c",
    "red-100": "#ffcdd2",
    "red-200": "#ef9a9a",
    "red-300": "#e57373",
    "red-400": "#ef5350",
    "red-50": "#ffebee",
    "red-500": "#f44336",
    "red-600": "#e53935",
    "red-700": "#d32f2f",
    "red-800": "#c62828",
    "red-900": "#b71c1c",
    "teal-100": "#b2dfdb",
    "teal-200": "#80cbc4",
    "teal-300": "#4db6ac",
    "teal-400": "#26a69a",
    "teal-50": "#e0f2f1",
    "teal-500": "#009688",
    "teal-600": "#00897b",
    "teal-700": "#00796b",
    "teal-800": "#00695c",
    "teal-900": "#004d40",
    "yellow-100": "#FFF9C4",
    "yellow-200": "#FFF59D",
    "yellow-300": "#FFF176",
    "yellow-400": "#FFEE58",
    "yellow-50": "#FFFDE7",
    "yellow-500": "#FFEB3B",
    "yellow-600": "#FDD835",
    "yellow-700": "#FBC02D",
    "yellow-800": "#F9A825",
    "yellow-900": "#F57F17",
  },
  "depp-purple-100": "#d1c4e9",
  "depp-purple-200": "#b39ddb",
  "depp-purple-300": "#9575cd",
  "depp-purple-400": "#7e57c2",
  "depp-purple-50": "#ede7f6",
  "depp-purple-500": "#673ab7",
  "depp-purple-600": "#5e35b1",
  "depp-purple-700": "#512da8",
  "depp-purple-800": "#4527a0",
  "depp-purple-900": "#311b92",
  "green-100": "#c8e6c9",
  "green-200": "#a5d6a7",
  "green-300": "#81c784",
  "green-400": "#66bb6a",
  "green-50": "#e8f5e9",
  "green-500": "#4caf50",
  "green-600": "#43a047",
  "green-700": "#388e3c",
  "green-800": "#2e7d32",
  "green-900": "#1b5e20",
  "grey-100": "#f5f5f5",
  "grey-200": "#eeeeee",
  "grey-300": "#e0e0e0",
  "grey-400": "#bdbdbd",
  "grey-50": "#fafafa",
  "grey-500": "#9e9e9e",
  "grey-600": "#757575",
  "grey-700": "#616161",
  "grey-800": "#424242",
  "grey-900": "#212121",
  "indigo-100": "#c5cae9",
  "indigo-200": "#9fa8da",
  "indigo-300": "#7986cb",
  "indigo-400": "#5c6bc0",
  "indigo-50": "#e8eaf6",
  "indigo-500": "#3f51b5",
  "indigo-600": "#3949ab",
  "indigo-700": "#303f9f",
  "indigo-800": "#283593",
  "indigo-900": "#1a237e",
  "light-blue-100": "#b3e5fc",
  "light-blue-200": "#81d4fa",
  "light-blue-300": "#4fc3f7",
  "light-blue-400": "#29b6f6",
  "light-blue-50": "#e1f5fe",
  "light-blue-500": "#03a9f4",
  "light-blue-600": "#039be5",
  "light-blue-700": "#0288d1",
  "light-blue-800": "#0277bd",
  "light-blue-900": "#01579b",
  "light-green-100": "#dcedc8",
  "light-green-200": "#c5e1a5",
  "light-green-300": "#aed581",
  "light-green-400": "#9ccc65",
  "light-green-50": "#f1f8e9",
  "light-green-500": "#8bc34a",
  "light-green-600": "#7cb342",
  "light-green-700": "#689f38",
  "light-green-800": "#558b2f",
  "light-green-900": "#33691e",
  "lime-100": "#f0f4c3",
  "lime-200": "#e6ee9c",
  "lime-300": "#dce775",
  "lime-400": "#d4e157",
  "lime-50": "#f9fbe7",
  "lime-500": "#cddc39",
  "lime-600": "#c0ca33",
  "lime-700": "#afb42b",
  "lime-800": "#9e9d24",
  "lime-900": "#827717",
  "orange-100": "#ffe0b2",
  "orange-200": "#ffcc80",
  "orange-300": "#ffb74d",
  "orange-400": "#ffa726",
  "orange-50": "#fff3e0",
  "orange-500": "#ff9800",
  "orange-600": "#fb8c00",
  "orange-700": "#f57c00",
  "orange-800": "#ef6c00",
  "orange-900": "#e65100",
  "pink-100": "#f8bbd0",
  "pink-200": "#f48fb1",
  "pink-300": "#f06292",
  "pink-400": "#ec407a",
  "pink-50": "#fce4ec",
  "pink-500": "#e91e63",
  "pink-600": "#d81b60",
  "pink-700": "#c2185b",
  "pink-800": "#ad1457",
  "pink-900": "#880e4f",
  "purple-100": "#e1bee7",
  "purple-200": "#ce93d8",
  "purple-300": "#ba68c8",
  "purple-400": "#ab47bc",
  "purple-50": "#f3e5f5",
  "purple-500": "#9c27b0",
  "purple-600": "#8e24aa",
  "purple-700": "#7b1fa2",
  "purple-800": "#6a1b9a",
  "purple-900": "#4a148c",
  "red-100": "#ffcdd2",
  "red-200": "#ef9a9a",
  "red-300": "#e57373",
  "red-400": "#ef5350",
  "red-50": "#ffebee",
  "red-500": "#f44336",
  "red-600": "#e53935",
  "red-700": "#d32f2f",
  "red-800": "#c62828",
  "red-900": "#b71c1c",
  "teal-100": "#b2dfdb",
  "teal-200": "#80cbc4",
  "teal-300": "#4db6ac",
  "teal-400": "#26a69a",
  "teal-50": "#e0f2f1",
  "teal-500": "#009688",
  "teal-600": "#00897b",
  "teal-700": "#00796b",
  "teal-800": "#00695c",
  "teal-900": "#004d40",
  "yellow-100": "#FFF9C4",
  "yellow-200": "#FFF59D",
  "yellow-300": "#FFF176",
  "yellow-400": "#FFEE58",
  "yellow-50": "#FFFDE7",
  "yellow-500": "#FFEB3B",
  "yellow-600": "#FDD835",
  "yellow-700": "#FBC02D",
  "yellow-800": "#F9A825",
  "yellow-900": "#F57F17",
}
`;
